images/cfw/opt/x1plus/lib/python/x1plus/client/mqtt.py

```diff
@@ -0,0 +1,184 @@
+        if 'ServiceUnknown' in str(e):
+            logger.info("Service not available, check DBus service status.")
+        return None
+    except Exception as e:
+        logger.error(f"Unexpected error during subscription: {e}")
+        return None
+
+def unsubscribe_topic(topic):
+    logger.info(f"Attempting to unsubscribe from topic: {topic}")
+    try:
+        response = call(MQTT_DBUS_ADDRESS, 'MQTTUnsubscribe', {'topic': topic})
+        logger.info(f"Unsubscribed from topic: {topic} with response: {response}")
+        return response
+    except Exception as e:
+        logger.error(f"Failed to unsubscribe from topic {topic}: {e}")
+        return None
+
+def get_status():
+    return call(MQTT_DBUS_ADDRESS, 'MQTTStatus')
+
+def list_subscriptions():
+    return call(MQTT_DBUS_ADDRESS, 'MQTTShowSubs')
+
+###
+import x1plus.utils
+from .payloads import *
+
+sn = x1plus.utils.serial_number()
+
+def _cmd_status(args=None):
+    status = get_status()
+    if status:
+        print(json.dumps(status, indent=2))
+    else:
+        print("Failed to retrieve MQTT service status")
+        
+# Payload-formatting publisher
+# CLI example: x1plus mqtt publish gcode_line "M106 P1 S200"
+def _cmd_publish(args):
+    command_to_function = {
+        'get_version': (get_version, []),
+        'upgrade_confirm': (upgrade_confirm, []),
+        'consistency_confirm': (consistency_confirm, []),
+        'start_upgrade': (start_upgrade, ['url', 'module', 'version']),
+        'get_upgrade_history': (get_upgrade_history, []),
+        'print_action': (print_action, ['action']),
+        'ams_change_filament': (ams_change_filament, ['target', 'curr_temp', 'tar_temp']),
+        'ams_control': (ams_control, ['param']),
+        'unload_filament': (unload_filament, []),
+        'get_access_code': (get_access_code, []),
+        'ipcam_record_set': (ipcam_record_set, ['control']),
+        'ipcam_timelapse': (ipcam_timelapse, ['control']),
+        'xcam_control_set': (xcam_control_set, ['module_name', 'control', 'print_halt']),
+        'send_project': (send_project, ['plate_number', 'url']),
+        'gcode_file': (gcode_file, ['file_path']),
+        'gcode_line': (gcode_line, ['command'])
+    }
+
+    command = args.command
+    if command in command_to_function:
+        func, param_keys = command_to_function[command]
+        if len(args.payload) != len(param_keys):
+            print(f"Error: Expected {len(param_keys)} arguments for {command}, got {len(args.payload)}")
+            return
+
+        kwargs = dict(zip(param_keys, args.payload))
+
+        try:
+            payload = func(**kwargs)
+            print(payload)
+            topic = f"device/{sn}/request"
+            result = publish_message(topic,payload)
+            if result:
+                print(f"Successfully published to topic {topic}. Payload: {json.loads(payload)}")
+            else:
+                print("Failed to publish message")
```

Comment:
I mention some in my broader review comments, but I don't think this is the right approach.

Reply:
I've been trying to figure out what to do with this, and it's kinda morphed into something that's out of control. 

What I'm thinking is this whole option (_cmd_publish()) needs to be axed and replaced with CLI options that are more focused to the application. For example for gcode publication and file printing
```
x1plus print gcode_line "M420 S69"
x1plus print gcode_file /sdcard/test.gcode
```

These other things definitely should not be command line options, for multiple reasons